File: GoogleIt/src/main/java/com.aliucord.plugins/GoogleIt.java
package com.aliucord.plugins;

import android.content.Context;
import androidx.annotation.NonNull;

import com.aliucord.Http;
import com.aliucord.Main;
import com.aliucord.api.CommandsAPI;
import com.aliucord.entities.Plugin;
import com.aliucord.annotations.AliucordPlugin;
import java.util.*;

@SuppressWarnings("unused")
@AliucordPlugin
public class GoogleIt extends Plugin {

    public static final String TargetUrl = "https://google.it/search?q=";

    @Override
    public void start(Context context) {
        commands.registerCommand("GoogleIt", "Generates a google.it link.", Collections.singletonList(CommandsAPI.requiredMessageOption), args -> {
            String msg = (String) args.get("message");
            if (msg == null) return new CommandsAPI.CommandResult(msg);
            String UrlResult = null;
            try {
                UrlResult = new Http.QueryBuilder("https://google.it/search").append("q", msg).toString();
            } catch (Throwable throwable) {
                throwable.printStackTrace();
            }
            return new CommandsAPI.CommandResult(UrlResult);
        });
    }

    @Override
    public void stop(Context context) {
        commands.unregisterAll();
    }

}

File: GoogleIt/src/main/java/com.aliucord.plugins/GoogleIt.kt
package com.aliucord.plugins

import android.content.Context
//import androidx.annotation.NonNull

import com.aliucord.Http
import com.aliucord.api.CommandsAPI
import com.aliucord.entities.Plugin
import com.aliucord.annotations.AliucordPlugin
import com.discord.models.commands.ApplicationCommandOption
import com.discord.api.commands.ApplicationCommandType

@AliucordPlugin
class GoogleIt : Plugin() {

    override fun start(ctx: Context) {
        val query = listOf(
                ApplicationCommandOption(
                        ApplicationCommandType.STRING,
                        "query",
                        "The query to search",
                        null,
                        true,
                        false,
                        null,
                        null
                )
        )
        commands.registerCommand(
                "googleit",
                "Generates a google.it link",
                query
        )
        { ctx ->
            try {
                val url = Http.QueryBuilder("https://google.it/search")
                        .append("q", ctx.getString("query"))
                        .toString()
                CommandsAPI.CommandResult(url)
            } catch (throwable: Throwable) {
                throwable.printStackTrace()
                CommandsAPI.CommandResult("An error occured", null, false)
            }
        }
    }

    override fun stop(ctx: Context) = commands.unregisterAll()

}
